docs/rst/tutorials/core/discoveryserver/discoveryserver_complex.rst

```diff
@@ -0,0 +1,380 @@
+
+.. code-block:: bash
+
+    fastdds discovery --server-id 0 -t 192.168.1.165 -q 42100 -t 127.0.0.1 -q 42101
+
+The output should look similar to the following:
+
+.. code-block:: bash
+
+    ### Server is running ###
+    Participant Type:   SERVER
+    Security:           NO
+    Server ID:          0
+    Server GUID prefix: 44.53.00.5f.45.50.52.4f.53.49.4d.41
+    Server Addresses:   TCPv4:[192.168.1.165]:42100-42100
+                        TCPv4:[127.0.0.1]:42101-42101
+
+The output displays the ``server ID`` set, followed by the security and server GUID prefix.
+Server address ``TCPv4:[192.168.1.165]`` tells *Fast DDS* to listen on Wifi 192.168.1.165 interface with the ``42100`` connection port.
+Server address ``TCPv4:[127.0.0.1]`` tells *Fast DDS* to listen on localhost with the ``42101`` connection port.
+
+After setting up the Discovery Server, you need to launch four additional Docker containers, two in each host, that will act as clients.
+First, open two new terminals in each machine and run the following command to start each container in the same network:
+
+.. code-block:: bash
+
+    docker run \
+        -it \
+        --name <container_name> \
+        --privileged \
+        --net host \
+        --ipc host \
+        -e DISPLAY=$DISPLAY \
+        -v /tmp/.X11-unix:/tmp/.X11-unix \
+        ubuntu-vulcanexus:humble-desktop
+
+Replace ``<container_name>`` with a unique name for each container.
+Once the containers are running, and after sourcing the Vulcanexus environment within each container, the easiest way to configure the clients to point to the *Discovery Server* is by setting `Environment Variables <https://fast-dds.docs.eprosima.com/en/latest/fastdds/env_vars/env_vars.html>`__.
+There are two different environment variables that adapt to different needs:
+
+1. ``ROS_DISCOVERY_SERVER``:
+
+    This is the simplest option.
+    It allows you to specify the addresses of the Discovery Servers that clients should contact to discover other nodes. You can set this variable inside the container as follows:
+
+    .. code-block:: bash
+
+        export ROS_DISCOVERY_SERVER=<server_ip>:<port>"
+
+    Replace ``<server_ip>`` with the IP address of the Discovery Server and <port> with the connection port.
+
+2. ``FASTDDS_ENVIRONMENT_FILE``:
+
+    This variable allows you to specify the path to a JSON file that contains all the environment variables.
+    You can specify the IP addresses and ports of the Discovery Servers the clients should contact.
+    It provides more flexibility, as you can modify the locators (i.e., server addresses) dynamically at runtime.
+    To use this option, create a JSON file with the following structure:
+
+    .. code-block:: xml
+
+        {
+            "ROS_DISCOVERY_SERVER": "<server_ip>:<port>"
+        }
+
+    Save the file in a known location and set the environment variable to point to it:
+
+    .. code-block:: bash
+
+        export FASTDDS_ENVIRONMENT_FILE="<json_file_path>"
```

Comment:
Too much information for a tutorial. I'd go for the JSON option as is the one used in the tutorial.